### PR TITLE
Added support for eosio blockchains

### PR DESCRIPTION
**Pull request by sebastianmontero**

### CODE REVIEW

**Comment by oed** [comment]
What signature algorithm does EOS use? Secp256k1 or something different?

**Comment by sebastianmontero** [comment]
yes, secp256k1 and also supports secp256r1

**Comment by oed** [comment]
Ok, my main concern here would be that this change would add a forth library that verifies secp256k1 to this project. 
Thoughts on this @zachferland? Can we perhaps also add this dependency as optional?

**Comment by zachferland** [comment]
Overall this looks good! just a few questions 

looks like accounts in eosio are unique strings? and they resolve to public keys on chain? do they resolve only to a single pubkey or multiple? 

just wondering if there is any implementation where the extra network requests are not needed 

what are the most commonly used eosio wallets/signers, especially for use with browser apps? not to familiar with ecosystem, would like to try at least one out 

I can rebase this after to resolve the conflicts before merging

**Comment by sebastianmontero** [comment]
Hi,

Looks like accounts in eosio are unique strings? and they resolve to public
keys on chain? do they resolve only to a single pubkey or multiple?

*Yes, an account is a unique string and can be associated with multiple
keys.*

just wondering if there is any implementation where the extra network
requests are not needed.

*The network request is required to get the public keys associated with an
account.*

what are the most commonly used eosio wallets/signers, especially for use
with browser apps? not to familiar with ecosystem, would like to try at
least one out

*Scatter is one of the most popular: https://get-scatter.com/
<https://get-scatter.com/>. The implementation assumes that the wallets
provide a UAL
Authenticator(https://github.com/EOSIO/universal-authenticator-library
<https://github.com/EOSIO/universal-authenticator-library>) and as such
implement the following
interface: https://github.com/EOSIO/universal-authenticator-library/blob/master/src/User.ts
<https://github.com/EOSIO/universal-authenticator-library/blob/master/src/User.ts>.*

Regards,

File: src/blockchains/eosio.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  const accountID = new AccountID(account)
+  const {address, chainId} = accountID
+  try {
+
+      const success = await SigningTools.verifySignature({
+        chainId: chainId.reference,
+        account: address,
+        signature,
+        data: toPayload(message, accountID)
+      });
```
Comment (zachferland) [inline src/blockchains/eosio.ts:98]: whats the error behavior here? does it throw when signature verification fails or only for other errors?

**Comment by zachferland** [comment]
thanks, ok makes sense then 

added one more question, then ill resolve the conflicts, and can likely be merged today 👍

**Comment by sebastianmontero** [comment]
Only for other errors, mainly a network error.

**Comment by zachferland** [comment]
Going to merge here - https://github.com/ceramicnetwork/js-3id-blockchain-utils/pull/42

we maybe changing support here a bit in the near future, where each blockchain support implementation will be its own package instead. This is will allow you to just include the dependencies you want, but since the additional dependencies here are not too large, going merge anyways for now.

**Pull request merged**